### PR TITLE
Optimize concat, crop, pad

### PR DESCRIPTION
**Pull request by Adib234**
## Summary
- [X] I have read CONTRIBUTING.md to understand how to contribute to this repository :)

<Please summarize what you are trying to achieve, what changes you made, and how they acheive the desired result.>
I'm trying to speed up the following video augmentations: concat, crop, and pad.

Here are the times which I measured with time for i in (seq 5); python -m unittest augly.tests.video_tests.transforms.ffmpeg_test.TransformsVideoUnitTest.test_{respective_augmentation};; end

Runtime with changes for Concat: 8.9s (test finished about 5s)
Runtime without changes for Concat: 22.012s (test finished about 20s)

Runtime with changes for Crop: 4.53s (test finished about 1.5s)
Runtime without changes for Crop: 5.426s (test finished about 3s)

Runtime with changes for Pad: 6.4s (test finished about 3s)
Runtime without changes for Pad: 15.62s (test finished about 12s)

## Unit Tests
If your changes touch the `audio` module, please run all of the `audio` tests and paste the output here. Likewise for `image`, `text`, & `video`. If your changes could affect behavior in multiple modules, please run the tests for all potentially affected modules. If you are unsure of which modules might be affected by your changes, please just run all the unit tests.

### Audio
```bash
python -m unittest discover -s augly/tests/audio_tests/ -p "*"
```

### Image
```bash
python -m unittest discover -s augly/tests/image_tests/ -p "*_test.py"
# Or `python -m unittest discover -s augly/tests/image_tests/ -p "*.py"` to run pytorch test too (must install `torchvision` to run)
```

### Text
```bash
python -m unittest discover -s augly/tests/text_tests/ -p "*"
```

### Video
```bash
python -m unittest discover -s augly/tests/video_tests/ -p "*"
```

### All
```bash
python -m unittest discover -s augly/tests/ -p "*"
```

## Other testing

If applicable, test your changes and paste the output here. For example, if your changes affect the requirements/installation, then test installing augly in a fresh conda env, then make sure you are able to import augly & run the unit test

### CODE REVIEW

File: augly/video/augmenters/ffmpeg/concat.py [hunk 1]
```diff
@@ -30,27 +30,41 @@ def __init__(self, video_paths: List[str], src_video_path_index: int):
...
+        flat_inputs = [element for sublist in inputs for element in sublist]
+        scale_and_sar, maps = "", ""
+        for i in range(len(self.video_paths)):
+            scale_and_sar += f"[{i}:v]scale={self.width}:{self.height}[{i}v],[{i}v]setsar=ratio={self.sample_aspect_ratio}[{i}vf];"
+
+        for i in range(len(self.video_paths)):
+            maps += f"[{i}vf][{i}:a]"
+
+        rest_command = f"concat=n={len(self.video_paths)}:v=1:a=1[v][a]"
+        command = [
```
Comment (zpapakipos) [inline augly/video/augmenters/ffmpeg/concat.py:54]: I remember you were having trouble putting this command together; have you verified that it works with a variety of videos, putting the videos in different orders, different numbers of videos, etc?

Comment (Adib234) [reply augly/video/augmenters/ffmpeg/concat.py:54]: I've tested it with 2 videos and put them in different orders but I will try with more videos now and get back to you.

Comment (Adib234) [reply augly/video/augmenters/ffmpeg/concat.py:54]: So I've tested with three videos and compare it AugLy and it was the same

File: augly/video/augmenters/ffmpeg/concat.py [hunk 2]
```diff
@@ -30,27 +30,41 @@ def __init__(self, video_paths: List[str], src_video_path_index: int):
...
-                    stream.audio,
-                )
-            )
-
-        return ffmpeg.concat(*streams, v=1, a=1, n=len(streams) // 2), {"vsync": 2}
+        inputs = [["-i", video] for video in self.video_paths]
+        flat_inputs = [element for sublist in inputs for element in sublist]
+        scale_and_sar, maps = "", ""
+        for i in range(len(self.video_paths)):
+            scale_and_sar += f"[{i}:v]scale={self.width}:{self.height}[{i}v],[{i}v]setsar=ratio={self.sample_aspect_ratio}[{i}vf];"
```
Comment (jbitton) [inline augly/video/augmenters/ffmpeg/concat.py:48]: this line is > 90 cols -- can you split into multiple lines?

**Review by jbitton — APPROVED** [review]
overall lgtm! just one tiny nit :)

File: augly/video/augmenters/ffmpeg/concat.py (hunk 2 above)
Comment (zpapakipos) [reply augly/video/augmenters/ffmpeg/concat.py:48]:
```suggestion
            scale_and_sar += (
                f"[{i}:v]scale={self.width}:{self.height}[{i}v],[{i}v]setsar=ratio="
                f"{self.sample_aspect_ratio}[{i}vf];"
            )
```

**Pull request closed without merging**